src/gettod.c

```diff
@@ -463,26 +461,19 @@ int W32_CALL gettimeofday2 (struct timeval *tv, struct timezone *tz)
 
   if (has_8254)
   {
+    static uint64 last = 0;
     static time_t secs = 0;           /* seconds since midnight */
     uint64 usecs = microsec_clock();  /* usec day-clock */
 
-#if STEVES_PATCHES
-    secs = time (NULL);
-#else
-    static uint64 last = 0;
     if (secs == 0 || usecs < last)    /* not init or wrapped */
     {
       secs = time (NULL);
       secs -= (secs % (24*3600));
```

Comment:
Not 100% sure about this line.  Could it cause a jump at midnight?  Maybe that is what the patch was intended to fix.

Reply:
I think so.